### ISSUE TITLE
Invalid C# file created with new Library project

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#6640]
- Affects master branch
 - Creating a new Library or PCL project results in a class containing the following code which does not compile:

```
using System;
namespace NewProject
{
    public class MyClass.cs
    {
        public MyClass.cs()
        {
        }
    }
}
```

There may be other affected projects.

I suspect this is related to the new file template change - https://github.com/mono/monodevelop/pull/6590

> VS bug [#733182](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/733182)

**Comment by mrward** [comment]
The new file template fix has broken how `${Name}` works in file templates. Before this change `${Name}` was the name of the new file without any file extension. Now it includes the file extension entered by the user.

Another example of this problem is to create a new Empty C# class library and enter 'Test.cs' rather than just 'Test'. Entering 'Test.cs' and then the `${Name}` parameter is replaced with 'Test.cs'.

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-13] Comment by **xamarinc**



Fixed in version 7.8.0.1329 (master)



Pull Request #6676 merged by: Matt Ward  
 Author: mono  
 Commit: f6bba9e3a7b85a76bebcc5f8975953133708a142 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-14] Comment by **xamarinc**



Fixed in version 7.8.0.1339 (release-7.8)



Pull Request #6676 merged by: Matt Ward  
 Author: mono  
 Commit: f6bba9e3a7b85a76bebcc5f8975953133708a142 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-17] Comment by **xamarinc**



Fixed in version 8.0.0.1738 (release-8.0)



Author: Matt Ward  
 Commit: 5b86ce0ef54a64bded1a95cbb958060d4038080f (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-19] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #6676 merged by: Matt Ward  
 Author: mono  
 Commit: f6bba9e3a7b85a76bebcc5f8975953133708a142 (xamarin/monodevelop)
